ui/app/hooks/useTransactionTimeRemaining.js

```diff
@@ -0,0 +1,54 @@
+import { getEstimatedGasPrices, getEstimatedGasTimes, getFeatureFlags } from '../selectors'
+import { hexWEIToDecGWEI } from '../helpers/utils/conversions.util'
+import { useSelector } from 'react-redux'
+import { useRef, useEffect, useState, useMemo } from 'react'
+import { isEqual } from 'lodash'
+import { getRawTimeEstimateData } from '../helpers/utils/gas-time-estimates.util'
+
+
+function calcTransactionTimeRemaining (initialTimeEstimate, submittedTime) {
+  const currentTime = (new Date()).getTime()
+  const timeElapsedSinceSubmission = (currentTime - submittedTime) / 1000
+  const timeRemainingOnEstimate = initialTimeEstimate - timeElapsedSinceSubmission
+
+  const renderingTimeRemainingEstimate = `~ ${Math.round(timeRemainingOnEstimate / 60)}min`
```

Comment:
This `min` string is not localized. I think this entire string will need to be localized in order to support other locales properly 🤔 

We could use this internationalization API instead: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat/format
But that would restrict us somewhat (we'd be stuck showing "1 min. ago", "this minute", or "in 1 min." for values of -1, 0, and 1 respectively, with no way to format it differently).

Reply:
This API is pretty cool though! It'd be nice to automatically support all locales without any transaction effort.

Reply:
I've discussed this with @rachelcope, and she agreed that the "narrow" relative time format would be OK to use here. It's not much longer.

Reply:
We would need a polyfill though. Our minimum supported Chrome and Firefox versions are much older than the introduction of this API.